packages/zen-bull-netting/src/ZenBullNetting.sol

```diff
@@ -484,10 +545,207 @@ contract ZenBullNetting is Ownable, EIP712, FlashSwap {
+            usdcToBorrow = bullToMint * IEulerSimpleLens(eulerLens).getDTokenBalance(usdc, zenBull)
+                / bullTotalSupply;
+        }
+
+        // deposit into crab
+        uint256 wethFromAuction = IWETH(weth).balanceOf(address(this));
+        IWETH(weth).withdraw(wethFromAuction);
+
+        ICrabStrategyV2(crab).deposit{value: ethIntoCrab}();
+
+        // flashswap WETH for USDC debt, and deposit crab + wethToLend into ZenBull
+        uint256 minWethForUsdcDebt =
+            wethToLend - (_params.depositsToProcess + wethFromAuction - ethIntoCrab);
+        _exactInFlashSwap(
+            usdc,
+            weth,
+            _params.wethUsdcPoolFee,
+            usdcToBorrow,
+            minWethForUsdcDebt,
+            uint8(FLASH_SOURCE.DEPOSIT_AUCTION),
+            abi.encodePacked(_params.crabAmount, wethToLend)
+        );
+
+        MemoryVar memory memVar;
+        memVar.remainingEth =
+            address(this).balance - (initialEthBalance - _params.depositsToProcess);
+
+        if (memVar.remainingEth > 0 && _params.flashDepositEthToCrab > 0) {
+            IFlashZen.FlashDepositParams memory params = IFlashZen.FlashDepositParams({
+                ethToCrab: _params.flashDepositEthToCrab,
+                minEthFromSqth: _params.flashDepositMinEthFromSqth,
+                minEthFromUsdc: _params.flashDepositMinEthFromUsdc,
+                wPowerPerpPoolFee: _params.flashDepositWPowerPerpPoolFee,
+                usdcPoolFee: _params.wethUsdcPoolFee
+            });
+
+            IFlashZen(flashZenBull).flashDeposit{value: memVar.remainingEth}(params);
+        }
+
+        // send oSqth to market makers
+        memVar.oSqthBalance = IERC20(oSqth).balanceOf(address(this));
+        for (uint256 j = 0; j < _params.orders.length; j++) {
+            if (_params.orders[j].quantity < memVar.oSqthBalance) {
+                memVar.oSqthBalance -= _params.orders[j].quantity;
+
+                IERC20(oSqth).transfer(_params.orders[j].trader, _params.orders[j].quantity);
+                emit DepositAuction(
+                    _params.orders[j].trader,
+                    _params.orders[j].bidId,
+                    _params.orders[j].quantity,
+                    _params.clearingPrice
+                    );
+            } else {
+                IERC20(oSqth).transfer(_params.orders[j].trader, memVar.oSqthBalance);
+                emit DepositAuction(
+                    _params.orders[j].trader,
+                    _params.orders[j].bidId,
+                    memVar.oSqthBalance,
+                    _params.clearingPrice
+                    );
+                break;
+            }
+        }
+
+        // send ZenBull to depositor
+        memVar.remainingDeposits = _params.depositsToProcess;
+        uint256 k = depositsIndex;
+        memVar.currentZenBullBalance =
+            IERC20(zenBull).balanceOf(address(this)) - initialZenBullBalance;
+        memVar.remainingEth =
```

Comment:
Already have this on 628

Reply:
Yeah but could change after the flashdeposit